packages/headless/src/state/insight-app-state.ts

```diff
@@ -28,4 +30,6 @@ export type InsightAppState = InsightSearchParametersState &
   ResultPreviewSection &
   SearchSection &
   QuerySetSection &
-  QuerySuggestionSection;
+  QuerySuggestionSection &
+  CategoryFacetSection &
```

Comment:
The insight `executeSearch` action (i.e., `buildInsightSearchRequest` function) will need to be modified as well to include the category facets in the search request parameters.